DOCS.md

```diff
@@ -0,0 +1,367 @@
+  **Default:** `true`
+
+- **`customVisitors?: Record<string, VisitorFn>`**
+  
+  Custom AST visitor functions for additional transformations during the compilation process.
+  
+  **Note:** This property is not currently defined in the TypeScript interface but is used in the implementation.
+
+- **`importFunction?: (source: string) => Object | Promise<Object>`**
+  
+  Custom import handler for resolving module imports. Can be synchronous or asynchronous.
+  
+  **Note:** If an async function is provided, `options.isAsync` will automatically be set to `true`.
+
+- **`isAsync?: boolean`**
+  
+  Wraps the generated code in an async function, enabling top-level await usage.
+  
+  **Default:** `false`
+
+### Return Value
+
+Returns a `CompiledCode` object with the following properties:
+
+```typescript
+interface CompiledCode {
+  origCode: string;    // The original input code
+  genCode: string;     // The generated/transformed JavaScript code
+  fn: Function;        // The compiled function ready for execution
+  args: any[];         // The arguments array to pass to the function
+  run: () => any;      // Convenience method to execute the compiled code
+}
+```
+
+#### Properties
+
+- **`origCode: string`** - The original input code passed to the compile function
+- **`genCode: string`** - The transformed JavaScript code generated by eval-magic
+- **`fn: Function`** - The compiled function ready for execution
+- **`args: any[]`** - The array of arguments that will be passed to the function when executed
+- **`run: () => any`** - A convenience method that executes the compiled function with the prepared arguments
+
+## Usage Examples
+
+### Basic Compilation
+
+```javascript
+import { compile } from "eval-magic";
+
+const compiled = compile(
+  `export const result = x + y;`,
+  { x: 10, y: 20 }
+);
+
+console.log(compiled.origCode);  // `export const result = x + y;`
+console.log(compiled.genCode);   // Generated JavaScript code
+console.log(compiled.run());     // { result: 30 }
+```
+
+### Using Return Instead of Export
+
+```javascript
+import { compile } from "eval-magic";
+
+const compiled = compile(
+  `return x * 2;`,
+  { x: 15 },
+  { returns: "return" }
+);
+
+console.log(compiled.run()); // 30
+```
+
+### Working with Complex Expressions
+
+```javascript
+import { compile } from "eval-magic";
+
+const compiled = compile(
+  `
+  const doubled = x * 2;
+  const squared = doubled ** 2;
+  export const result = squared + offset;
+  `,
+  { x: 5, offset: 10 }
+);
+
+console.log(compiled.run()); // { result: 110 }
+```
+
+### Operator Overloading
+
+```javascript
+import { compile, Py } from "eval-magic";
+
+class Point {
+  constructor(x, y) {
+    this.x = x;
+    this.y = y;
+  }
+
+  [Py.__add__](other) {
+    return new Point(this.x + other.x, this.y + other.y);
+  }
+
+  toString() {
+    return `Point(${this.x}, ${this.y})`;
+  }
+}
+
+const compiled = compile(
+  `export const result = p1 + p2;`,
+  {
+    p1: new Point(1, 2),
```

Comment:
a better example is to pass the `Point` class to the scope, and define the p1 and p2 vars inside the code it self!